pkg/controller/cluster/cidr_allocation.go

```diff
@@ -0,0 +1,118 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/galal-hussein/k3k/pkg/apis/k3k.io/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	cidrAllocationClusterPoolName = "k3k-cidr-allocation-cluster-pool"
```

Comment:
can we slightly change the name to match the k3s flags: cluster-cidr and service-cidr instead of cidr-allocation-cluster